lib/pbench/server/__init__.py

```diff
@@ -186,7 +187,7 @@ def COMMIT_ID(self) -> str:
 
     @property
     def rest_uri(self) -> str:
-        return self._get_conf("pbench-server", "rest_uri")
+        return self.conf.get("pbench-server", "rest_uri")
```

Comment:
This represents a change of the `rest_uri` interface -- it will now report errors via `NoOptionError`/`NoSectionError` instead of `BadConfig` -- are we OK with that?

(I don't see any changes to the "callers" of `rest_uri` in this PR...do they just assume that it won't fail and/or one error is just as good as another from the callers' points of view?)

Reply:
It's called in `endpoint_configure.py`; and an exception, which is a highly extraordinary event, will result in Flask default `Exception` behavior and a 500 error. That's probably OK in this context given the nature of the configuration problem we're encountering. But the definition of our rest URI point is one of those things that ought to be validated before proper server startup in which case it wouldn't matter.

Reply:
True (and thanks for the reassurance, Dave), but I just want to make sure that, after all of the recent efforts on Peter's part to switch from raising `NoOptionError`/`NoSectionError` to raising `BadConfig`,  he's aware that this is moving in the opposite direction.  (I'm not opposed to this change, as long as it is understood/intentional.)

Reply:
I've added it to the list of required properties.